spec/lib/appsignal/heartbeat_spec.rb

```diff
@@ -0,0 +1,62 @@
+describe Appsignal::Heartbeat do
+  let(:config) { project_fixture_config }
+  let(:heartbeat) { described_class.new(:name => "heartbeat-name") }
+  let(:transmitter) { Appsignal::Transmitter.new("http://heartbeats/", config) }
+
+  before(:each) do
+    config.logger = Logger.new(StringIO.new)
+    allow(Appsignal::Heartbeat).to receive(:transmitter).and_return(transmitter)
+  end
+
+  describe "#start" do
+    it "should send a heartbeat start" do
+      expect(transmitter).to receive(:transmit).with(hash_including(
+        :name => "heartbeat-name",
+        :kind => "start"
+      )).and_return(nil)
+
+      heartbeat.start
+    end
+  end
+
+  describe "#finish" do
+    it "should send a heartbeat finish" do
+      expect(transmitter).to receive(:transmit).with(hash_including(
+        :name => "heartbeat-name",
+        :kind => "finish"
+      )).and_return(nil)
+
+      heartbeat.finish
+    end
+  end
+
+  describe ".heartbeat" do
+    describe "when a block is given" do
+      it "should send a heartbeat start and finish and return the block output" do
```

Comment:
Would be good to add a test case for when an error occurs in the block, that it doesn't send the finish event.

Reply:
Ooh, yes, absolutely. Thank you for spotting that! 💯